### PR TITLE
Change getattr to xgetattr in tree get children

### LINKED ISSUE #491: Tree node get children method should handle nested trait names
The [`get_children`](https://github.com/enthought/traitsui/blob/193898594c85ad612b8b7ab9815bc810ace466f8/traitsui/tree_node.py#L219-L222) method uses the standard `getattr` rather than `traits.trait_base.xgetattr`, and so nested trait names cannot be used as children.

### PR DESCRIPTION
**Pull request by notmatthancock**
Closes #491

### CODE REVIEW

File: traitsui/tree_node.py [hunk 1]
```diff
@@ -246,7 +247,8 @@ def append_child(self, object, child):
     def insert_child(self, object, index, child):
         """ Inserts a child into the object's children.
         """
-        getattr(object, self.children)[index: index] = [child]
+        children = self.get_children()
+        children[index: index] = [child]
```
Comment (rkern) [inline traitsui/tree_node.py]:
```suggestion
        children[index:index] = [child]
```

**Comment by notmatthancock** [comment]
Updated the comment. Note sure why travis is failing

**Comment by corranwebster** [comment]
Travis failure look to be related to PySide2, which is new to the test suite (as in its been there about 3 months), and it seems to have started dying before any tests actually run.  The issue is orthogonal to this PR.

**Comment by corranwebster** [comment]
This is a semantic change, as the behaviour of classes that override `get_children` but not the other methods will change.  However, I think that any class that overrides `get_children` would have had to override the other methods as well, so I think that this shouldn't break anyone's code.

File: traitsui/tree_node.py [hunk 2]
```diff
@@ -219,7 +220,7 @@ def has_children(self, object):
     def get_children(self, object):
         """ Gets the object's children.
         """
-        return getattr(object, self.children)
+        return xgetattr(object, self.children)
```
Comment (corranwebster) [inline traitsui/tree_node.py]:
```suggestion
        return xgetattr(object, self.children, None)
```
The default return value of `xgetattr` is `Undefined` (see https://github.com/enthought/traits/blob/master/traits/trait_base.py#L345).  For backwards compatibility you need the default to be `None`.

**Comment by corranwebster** [comment]
If you are going to do this in these three cases, you should probably do elsewhere in the file as well, eg. https://github.com/enthought/traitsui/blob/ffeb70b99979da852292ad9b785b181e833e0eb1/traitsui/tree_node.py#L462 and https://github.com/enthought/traitsui/blob/ffeb70b99979da852292ad9b785b181e833e0eb1/traitsui/tree_node.py#L1737 and several other places.

**Comment by notmatthancock** [comment]
Suggestions incorporated. Thanks @corranwebster

**Pull request merged**